centralizedmetrics/providers/mixpanel.go

```diff
@@ -0,0 +1,192 @@
+	req.Header.Set("content-type", "application/json")
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		body, err := io.ReadAll(resp.Body)
+		fmt.Println(resp.StatusCode, string(body), err)
+		log.Warn("failed to send metric", "status-code", resp.StatusCode, "body", resp.Body)
+		return errors.New("failed to send metric to Mixpanel")
+	}
+
+	return nil
+}
+
+func toMixpanelMetric(metric common.Metric, token string) mixpanelMetric {
+
+	properties := mixpanelMetricProperties{
+		Time:                 metric.Timestamp,
+		UserID:               metric.UserID,
+		Platform:             metric.Platform,
+		InsertID:             metric.ID,
+		AppVersion:           metric.AppVersion,
+		Token:                token,
+		AdditionalProperties: metric.EventValue,
+	}
+
+	return mixpanelMetric{
+		Event:      metric.EventName,
+		Properties: properties,
+	}
+}
+
+type mixpanelMetricProperties struct {
+	Time                 int64                  `json:"time"`
+	UserID               string                 `json:"distinct_id"`
+	InsertID             string                 `json:"$insert_id"`
+	Platform             string                 `json:"platform"`
+	AppVersion           string                 `json:"app_version"`
+	AdditionalProperties map[string]interface{} `json:"-"`
+	Token                string                 `json:"token"`
+}
+
+type mixpanelMetric struct {
+	Event      string                   `json:"event"`
+	Properties mixpanelMetricProperties `json:"properties"`
+}
+
+func (mmp mixpanelMetricProperties) MarshalJSON() ([]byte, error) {
+	// Create a map and marshal the struct fields into it
+	type alias mixpanelMetricProperties // Alias to avoid recursion
+	data, err := json.Marshal(alias(mmp))
+	if err != nil {
+		return nil, err
+	}
+
+	// Unmarshal the JSON into a map
+	var mmpMap map[string]interface{}
+	if err := json.Unmarshal(data, &mmpMap); err != nil {
+		return nil, err
+	}
+
+	// Merge AdditionalProperties into the map
+	for key, value := range mmp.AdditionalProperties {
+		mmpMap[key] = value
+	}
+
+	// Marshal the merged map back to JSON
+	marshaled, err := json.Marshal(mmpMap)
+
+	return marshaled, err
+}
+
+func (mmp *mixpanelMetricProperties) UnmarshalJSON(data []byte) error {
+	// Create a temporary alias type to unmarshal known fields
+	type alias mixpanelMetricProperties
+	aux := &struct {
+		*alias
+	}{
+		alias: (*alias)(mmp),
+	}
+
+	if err := json.Unmarshal(data, &aux); err != nil {
+		return err
+	}
+
+	// Unmarshal into a map to capture additional properties
+	var rawMap map[string]interface{}
+	if err := json.Unmarshal(data, &rawMap); err != nil {
+		return err
+	}
+
+	// Remove known fields from the map
+	delete(rawMap, "time")
+	delete(rawMap, "token")
+	delete(rawMap, "distinct_id")
+	delete(rawMap, "$insert_id")
+	delete(rawMap, "platform")
+	delete(rawMap, "screen")
```

Comment:
`screen` is not present in the struct `mixpanelMetricProperties`. Is that intended?